src/controllers/user.controller.js

```diff
@@ -55,6 +56,40 @@ const login = async (req, res, next) => {
   )(req, res, next);
 };
 
+const loginWithGoogle = async (req, res, next) => {
+  let user;
+  try {
+    user = await userServices.getUserByEmail(req.user.email);
+    if (!user) {
+      const data = {
+        username: req.user.displayName,
+        email: req.user.email,
+        password: 'null',
+      };
+      user = await userServices.createUser(data);
```

Comment:
`username` could be email since many users can have same display name, so user logged in with email can have usernames of their email.